File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    let a=0;
    let b=0;
    const arr = polynomial.split(" + ");
    arr.forEach(el=>{
        if(el.includes("x")){
            const p=el.slice(0,-1);
            p?a+=Number(p):a+=1;
        }else{
            b+=Number(el);
        }
    })
    let answer ="";
    if(a>0 &&b>0) answer =`${a!==1?a:""}x + ${b}`;
    else if(a>0 &&b===0) answer = `${a!==1?a:""}x`;
    else if (a===0 && b>0) answer = `${b}`;
    return answer;
}

File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    answer =[]
    x=0
    n=0
    spl = polynomial.split(" + ")
    for i in spl:
        if "x" in i:
            if i[0]=="x":
                x+=1
            else:
                x+=int(i[0:-1])
        else:
            n+=int(i)
    if x>0:
        if x==1:
            answer.append("x")
        else:
            answer.append(str(x)+"x")
    if n>0:
        answer.append(str(n))
    return " + ".join(answer)